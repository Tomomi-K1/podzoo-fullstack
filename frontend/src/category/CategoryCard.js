import React, { useState, useEffect } from "react";
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Collapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import FavoriteIcon from '@mui/icons-material/Favorite';
import ShareIcon from '@mui/icons-material/Share';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import MoreVertIcon from '@mui/icons-material/MoreVert';


function CategoryCard({category}){
    console.log(`category:${category}`)
    return(
        <Card sx={{maxWidth: "200px", display:"inline-block"}} raised={true} className={category.id}>
            {/* <CardMedia
                component="img"
                height="200px"
                image={podcast.artwork}
                alt="podcast artwork"
            /> */}
                <CardContent>
                    <Typography variant="h6" color="text.secondary" sx={{fontWeight:'bold'}}>
                        {category.name}
                    </Typography>
                </CardContent>
        </Card>
    )
}

export default CategoryCard;