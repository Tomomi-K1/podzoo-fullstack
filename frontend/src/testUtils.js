import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };