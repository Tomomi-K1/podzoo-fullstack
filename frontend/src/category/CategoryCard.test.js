import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import CategoryCard from "./CategoryCard";
import { UserProvider } from "../testUtils";

const category ={
    id: 1,
    name: 'test category'
    }
it('renders', () => {
    render(
        <MemoryRouter>
            <CategoryCard category={category}/>
        </MemoryRouter>
    ) 
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <CategoryCard category={category}/>
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

it('has correctly renders', () => {
    const category ={
        id: 1,
        name: 'test category'
    }
    render(
        <MemoryRouter>
            <CategoryCard category={category} />
        </MemoryRouter>
        )
    expect(screen.getByText('test category')).toBeInTheDocument();
})
