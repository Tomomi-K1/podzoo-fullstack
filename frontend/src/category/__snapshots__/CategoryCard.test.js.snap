// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <a
      href="/search/test category"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
      >
        test category
      </p>
    </a>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
